geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManagerImpl.java

```diff
@@ -1058,25 +1056,30 @@ public synchronized String toString() {
       return sb.toString();
     }
 
-    public synchronized void addConnection(PooledConnection connection) {
-      addToEndpointMap(connection);
-
-      // we want the smallest birthDate (e.g. oldest cnx) at the front of the list
-      getPoolStats().incPoolConnections(1);
-      // logger.info("DEBUG: addConnection incPoolConnections(1)->" +
-      // getPoolStats().getPoolConnections() + " con="+connection,
-      // new RuntimeException("STACK"));
-      this.allConnections.addLast(connection);
-      if (isIdleExpirePossible()) {
-        startBackgroundExpiration();
+    public void addConnection(PooledConnection connection) {
+      if (this.closing) {
```

Comment:
This won't fix the issue, though it might help hide it. What if `this.closing` is set after we check but before we synchronize?